pkg/registry/apis/provisioning/repository/repository.go

```diff
@@ -111,8 +111,8 @@ type Repository interface {
 
 // Hooks called after the repository has been created, updated or deleted
 type RepositoryHooks interface {
-	OnCreate(ctx context.Context) (*provisioning.RepositoryStatus, error)
-	OnUpdate(ctx context.Context) (*provisioning.RepositoryStatus, error)
+	OnCreate(ctx context.Context) (*provisioning.WebhookStatus, error)
+	OnUpdate(ctx context.Context) (*provisioning.WebhookStatus, error)
```

Comment:
🤔 this means the callbacks are only for webooks -- that is fine if the names are not so generic.  Maybe it should be called `RepositoryWebhooks` or similar?

Reply:
yeah,  I had the same thought. Let me think about it in another refactoring PR now that we can use patch